docs/src/man/comparisons.md

```diff
@@ -1,77 +1,200 @@
 # Comparisons
 
-This section compares DataFrames.jl with other data manipulation frameworks.
+This section compares DataFrames.jl with other data manipulation frameworks in Python, R, and Stata.
+
+A sample data set can be created using the following code:
+
+```julia
+df = DataFrame(id = 'a':'f', grp = repeat(1:2, 3), x = 6:-1:1, y = 4:9, z = [3:7; missing])
+df2 = DataFrame(grp = 1, w = 10)
+```
+
+## Comparison with the Python package pandas
+
+The following table compares the main functions of DataFrames.jl with the Python package pandas (version 1.1.0):
+
+```python
+df = pandas.DataFrame({'grp': [1, 2, 1, 2, 1, 2],
+                       'x': range(6, 0, -1),
+                       'y': range(4, 10),
+                       'z': [3, 4, 5, 6, 7, None]},
+                      index = list('abcdef'))
+
+>>> df
+   grp  x  y    z
+a    1  6  4  3.0
+b    2  5  5  4.0
+c    1  4  6  5.0
+d    2  3  7  6.0
+e    1  2  8  7.0
+f    2  1  9  NaN
+```
+
+By comparison, this pandas data frame has `a` to `f` as row indices rather than a separate `id` column.
+In the DataFrames.jl examples we use `findfirst` function as row index in pandas
+in general allows duplicates and in such case a row corresponding to the first
+occurrence of a value in the row index is returned.
+In DataFrames.jl in general a `groupby` function should be used to set a row index
+to a `DataFrame`; opposed to pandas it then returns all rows matching an index in
+case of duplicates.
```

Comment:
My favorite hobby is to switch "a" and "the" in @bkamins's texts. :-p
```suggestion
In the DataFrames.jl examples we use the `findfirst` function, since row index in pandas
in general allows duplicates and in such case the row corresponding to the first
occurrence of the value in the row index is returned.
In DataFrames.jl in general the `groupby` function should be used to set a row index
to a `DataFrame`; contrary to pandas it then returns all rows matching an index in
case of duplicates.
```
But more seriously, is it right that only the first row is returned when there are duplicates, given what you showed in the other thread?

Reply:
I haven't paid much attention about the duplicates before. Now, I just did an exercise with the last row's id changed to an 'e'.  Here're the findings for the cases 1) row indexing by label and 2) mixed indexing:

pandas:
```
>>> df
   grp  x  y    z
a    1  6  4  3.0
b    2  5  5  4.0
c    1  4  6  5.0
d    2  3  7  6.0
e    1  2  8  7.0
e    2  1  9  NaN
>>> df.loc['e']
   grp  x  y    z
e    1  2  8  7.0
e    2  1  9  NaN
>>> df.loc['e'][1]
Traceback (most recent call last):
  File "/home/ec2-user/miniconda3/envs/pandas/lib/python3.8/site-packages/pandas/core/indexes/base.py", line 2889, in get_loc
    return self._engine.get_loc(casted_key)
  File "pandas/_libs/index.pyx", line 70, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/index.pyx", line 97, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/hashtable_class_helper.pxi", line 1675, in pandas._libs.hashtable.PyObjectHashTable.get_item
  File "pandas/_libs/hashtable_class_helper.pxi", line 1683, in pandas._libs.hashtable.PyObjectHashTable.get_item
KeyError: 1
```

Two observations:
1. The row indexing by label example returns a data frame with all rows that matched the label.
2. The mixed indexing example is "unsafe" as the return type is unstable.

Now, compare to this Julia code, which always returns a DataFrameRow.
```
df[findfirst(==('c'), df.id), :]
```

I would like to propose two changes from here:

1) Move this paragraph to the section under "Accessing data" after the note about 1-based indexing.

2) Rewrite it again as follows. I want to focus on the Julia solution more than explaining what pandas does.

In the DataFrames.jl examples, the `findfirst` function is used to find the first match and return the result
as a single DataFrameRow object. In the case that `id` is not unique, you can use the `findall` function 
instead, and it would return a DataFrame object having all matched rows.

Reply:
BTW, I also did a little benchmarking comparing `findall` vs `groupby`, and I found `findall` performs better. That's why I am suggesting that in the above text. Please let me know if I miss anything.

```
julia> df2 = DataFrame(id=rand('a':'e', 10000), grp=1:10000, x=1:10000);

julia> foo1(df) = df[findall(==('e'), df.id), :];

julia> foo2(df) = groupby(df, :id)[('e',)];

julia> @btime foo1($df2);
  62.421 μs (27 allocations: 73.41 KiB)

julia> @btime foo2($df2);
  109.985 μs (50 allocations: 381.58 KiB)
```

Reply:
> My favorite hobby is to switch "a" and "the" in @bkamins's texts.

I should start reporting in Julia surveys that I consider myself a member of an underrepresented community of people whose native language does not have articles :smile:.

Reply:
@tk3369 `groupby` is supposed to be slower than `findall` if you do only one operation. `groupby` is for cases when you do it once and then do lookup many times.

Reply:
Now - you are right and my suggestion was wrong. I do not know why my previous test has shown that only the first occurrence. Indeed pandas returns all occurrences, but with the distinction that if there is only one then it drops a data frame. So my suggestion is wrong and should not be applied.

Reply:
regarding noting `findall(==('c'), df.col)` vs `df.col .== 'c'` I think both are OK, but probably the latter is easier to write.

Reply:
Thanks. I have now applied the change. Please review the latest commit.